src/Essentials/src/FileSystem/FileSystemUtils.uwp.cs

```diff
@@ -0,0 +1,52 @@
+﻿#nullable enable
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using Microsoft.Maui.ApplicationModel;
+using Package = Windows.ApplicationModel.Package;
+
+namespace Microsoft.Maui.Storage
+{
+	static partial class FileSystemUtils
+	{
+		public static bool AppPackageFileExists(string filename)
+		{
+			var file = PlatformGetFullAppPackageFilePath(filename);
+			return File.Exists(file);
+		}
+
+		public static string PlatformGetFullAppPackageFilePath(string filename)
+		{
+			if (filename is null)
+				throw new ArgumentNullException(nameof(filename));
+
+			filename = NormalizePath(filename);
+
+			string root;
+			if (AppInfoUtils.IsPackagedApp)
+				root = Package.Current.InstalledLocation.Path;
+			else
+				root = AppContext.BaseDirectory;
+
+			return Path.Combine(root, filename);
+		}
+
+		public static bool TryGetAppPackageFileUri(string filename, [NotNullWhen(true)] out string? uri)
+		{
+			var path = PlatformGetFullAppPackageFilePath(filename);
+
+			if (File.Exists(path))
+			{
+				if (AppInfoUtils.IsPackagedApp)
+					uri = $"ms-appx:///{filename.Replace('\\', '/')}";
+				else
+					uri = $"file:///{path.Replace('\\', '/')}";
```

Comment:
This can be handled by the bcl in a much nicer way: 

```csharp
var uri = new Uri("c:\\foo");
var converted = uri.AbsoluteUri;
```

Example:

```
sharp> var r = new Uri ("C:\\foo\\bar")
csharp> r.AbsolutePath                                                                                                                                                               "C:/foo/bar"                                                                                                                                                                         csharp> r.AbsoluteUri                                                                                                                                                                "file:///C:/foo/bar"                                                                                                                                                                 csharp>
```

No need to worry about corner cases etc